cuqi/experimental/mcmc/_gibbs.py

```diff
@@ -121,6 +129,14 @@ def __init__(self, target: JointDistribution, sampling_strategy: Dict[str, Sampl
         # Store parameter names
         self.par_names = self.target.get_parameter_names()
 
+        # Store the scan order
+        self._scan_order = scan_order
+
+        # Check that the parameters of the target align with the sampling_strategy and scan_order
+        for par_name in self.par_names:
+            if par_name not in self.scan_order:
```

Comment:
can may be use condition like
```
if set(self.par_names) != set(self._scan_order):
    #raise Error
```
to ensure scan order is not only a superset, but equal to the par_names